src/stan/callbacks/unique_stream_writer.hpp

```diff
@@ -77,6 +91,20 @@ class unique_stream_writer final : public writer {
       return;
     write_vector(values);
   }
+  /**
+   * Writes a set of values in csv format followed by a newline.
+   *
+   * Note: the precision of the output is determined by the settings
+   *  of the stream on construction.
+   *
+   * @param[in] values Values in a std::vector
+   * @param[in, out] ss ignored
+   */
+  void operator()(const std::vector<double>& values, std::stringstream& ss) {
```

Comment:
I don't think I like these overloads --

1. They're not in the base class `writer`
2. Is there a reason we cant just have the buffer be a private member in the class or something similar?
3. The "Note" in the doc comment is wrong, since we make no attempt to make sure the settings of the stringstream are the same as the ostream. I think, among other things, this would break the way cmdstan sets the number of digits?

Separately, isn't `std::ostream` buffered internally anyway?

Reply:
> Is there a reason we cant just have the buffer be a private member in the class or something similar?

I didn't want to introduce state directly in the class

> The "Note" in the doc comment is wrong, since we make no attempt to make sure the settings of the stringstream are the same as the ostream. I think, among other things, this would break the way cmdstan sets the number of digits?

I totally missed this. Yeah my through was that having a little buffer we reuse would save some memory, but idt it's worth the overhead of config etc. And we only use it for the case of a std vector really